### PR TITLE
Update storybook

### PR DESCRIPTION
**Pull request by coling**
This is an attempt to update story book to the latest version. Comments welcome.

## Description

I updated my dev environment to npm 7 and ran into problems with storybook build so attempted to fix this up. I suspect the changes could all be avoided just by using `npm install --legacy-peer-deps`  which I ultimately had to use anyway but I figure these changes will be needed eventually anyway.

Sadly, there are a bunch of problems that still remain (namely that this project requires react 17 and storybook is sadly still stuck requiring react 16 due to it's use of @reach/router. There are plans afoot to fix [this](https://github.com/storybookjs/storybook/issues/14619), but in the mean time I still had to use `npm install --legacy-peer-deps` to get a working build.

In Storybook 6, [addon-knobs has been deprecated](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#deprecated-addon-knobs) but it still works for now. It should ultimately be migrated to addon-controls

## Motivation and Context

I wanted this to work with latest npm. Failed and still had to use `--legacy-peer-deps` but upgrade still worked as a result.

## How Has This Been Tested?

I've tested this with npm, not used yarn. I was able to build the storybook and run it directly. I've not tried anything else related to release or building of the project.

Other than that, this was latest npm/node packages from npmsource DNF repository running in a CentOS 8 container.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Development/Build/Documentation change

### CODE REVIEW

**Comment by coling** [comment]
Just as a general comment, the last pull request I made with multiple commits was ultimately squashed and committed as one commit rather than merged. Don't know if this is a standard convention but it was quite confusing to see this happen as the detail in the individual commits was lost. IMO merging the branch (possibly with --no-ff) maintains a better history but I'm sure this is generally down to personal preference :-)

Hope you're well? Cheers!

**Review by coling — COMMENTED** [review]
For what it's worth, I found out why this was an issue. It's due to [an upstream bug](https://github.com/storybookjs/storybook/issues/13068) which although marked as closed, it's just been marked as a duplicate. This workaround here seems like a valid approach for now.

**Comment by samsam-ahmadi** [comment]
Hey Thanks for you PR. 

I'm feeling good, Thanks. 
Sorry about the squashed your last PR, I was thinking maybe it would be clean the history but i was wrong. 
I will merge these PRs as before. 
Just one comment about moving the react to the devdepenceny It's make some error when we want to link the component in local and use it in another project.  
Would you please double check ?

File: package.json [hunk 1]
```diff
@@ -21,16 +21,12 @@
     "rangePicker",
     "datePicker"
   ],
-  "peerDependencies": {
-    "react": "^17.0.1",
```
[the remark this answers is not in the document]
Comment (coling) [reply package.json:25]: Ahh perhaps I misunderstood the docs... They suggested adding dev deps which I incorrectly assumed meant peerDeps weren't needed. Keeping them as both should work fine I guess... Will have a play later.

[the remark this answers is not in the document]
Comment (samsam-ahmadi) [reply package.json:25]: Ahh, Sorry Maybe I was wrong, Please have a look at [this](https://github.com/facebook/react/issues/14257);

I had double react version problem at the beginning of this package, when I was using react in dev deps, I search a lot then some guys suggest me to using in the pear deps. after the new version, it has not any problem after install it in the project.

**Pull request merged**